### PR TITLE
Add rolemenu support to V3

### PR DESCRIPTION
**Pull request by VelvetToroyashi**
This PR exists for logging and visibility, as well as code-review. V3 at this time is unfinished, and will be a seperate PR, hence the target branch.

### CODE REVIEW

File: plugins/RoleMenuPlugin/Database/MediatR/GetRoleMenuRequest.cs [hunk 1]
```diff
@@ -12,23 +13,23 @@ public static class GetRoleMenu
         ///     Retrieves a <see cref="RoleMenuModel" /> from the database in the form of a <see cref="RoleMenuModel" />
         /// </summary>
         /// <param name="MessageId">The Id of the message to grab.</param>
-        public sealed record Request(ulong MessageId) : IRequest<RoleMenuModel>;
+        public sealed record Request(ulong MessageId) : IRequest<Result<RoleMenuModel>>;
```
Comment (CalebABG) [inline plugins/RoleMenuPlugin/Database/MediatR/GetRoleMenuRequest.cs:16]: Removing the `Request` suffix from the classes within the `MediatR` folder for the RoleMenuPlugin would make the naming consistent for the classes within the folder (not a big deal tho, just something small)

File: plugins/RoleMenuPlugin/temp.txt [hunk 2]
```diff
@@ -0,0 +1,54 @@
+sb
```
Comment (CalebABG) [inline plugins/RoleMenuPlugin/temp.txt:1]: Is file still needed xD ?

Comment (VelvetToroyashi) [reply plugins/RoleMenuPlugin/temp.txt:1]: Git

Comment (VelvetToroyashi) [reply plugins/RoleMenuPlugin/temp.txt:1]: Please.

**Review by CalebABG — APPROVED** [review]
LGTM 🚀

**Pull request merged**